x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/query_bar/index.tsx

```diff
@@ -168,6 +182,38 @@ export const QueryBarDefineRule = ({
     [field.value]
   );
 
+  const onCloseTimelineModal = useCallback(() => {
+    setLoadingTimeline(true);
+    onCloseTimelineSearch();
+  }, []);
+
+  const onOpenTimeline = useCallback(
+    (timeline: TimelineModel) => {
+      setLoadingTimeline(false);
+      const newQuery = {
+        query: timeline.kqlQuery.filterQuery?.kuery?.expression ?? '',
+        language: timeline.kqlQuery.filterQuery?.kuery?.kind ?? 'kuery',
+      };
+      const dataProvidersDsl =
+        timeline.dataProviders != null && timeline.dataProviders.length > 0
+          ? convertKueryToElasticSearchQuery(
+              buildGlobalQuery(timeline.dataProviders, browserFields),
+              indexPattern
+            )
+          : '';
+      const newFilters = timeline.filters ?? [];
+      field.setValue({
+        filters:
+          dataProvidersDsl !== ''
+            ? [...newFilters, getDataProviderFilter(dataProvidersDsl)]
+            : newFilters,
+        query: newQuery,
+        saved_id: '',
+      });
+    },
+    [browserFields, indexPattern]
```

Comment:
Additional deps -- `field` is coming from component props

```suggestion
    [browserFields, field, indexPattern]
```